src/farcasterd/trade_state_machine.rs

```diff
@@ -1171,6 +1193,22 @@ fn attempt_transition_from_restoring_swapd_to_swapd_running(
             runtime.handle_failed_connection(event.endpoints, source.clone())?;
             expect_connection = false;
         }
+        (BusMsg::Ctl(CtlMsg::Hello), source) if trade_role == TradeRole::Maker => {
+            if let Some(node_id) = expected_counterparty_node_id {
+                if source.node_addr()
+                    == Some(NodeAddr::new(
+                        node_id,
+                        runtime.config.get_bind_addr().unwrap(),
```

Comment:
What about making the `get_bind_addr` non-failing. If the config is found but errors we return default values. That would also fix the issue above.

Reply:
I think it's good to handle this error and let the user know of the broken config.

Reply:
f71102d758af53c12fd843ca2f20eef87bc6701e